src/js/control-bar/volume-panel.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * @file volume-control.js
+ */
+import Component from '../component.js';
+import checkVolumeSupport from './volume-control/check-volume-support';
+import {isPlain} from '../utils/obj';
+
+// Required children
+import './volume-control/volume-control.js';
+import './mute-toggle.js';
+
+/**
+ * A Component to contain the MuteToggle and VolumeControl so that
+ * they can work together.
+ *
+ * @extends Component
+ */
+class VolumePanel extends Component {
+
+  /**
+   * Creates an instance of this class.
+   *
+   * @param {Player} player
+   *        The `Player` that this class should be attached to.
+   *
+   * @param {Object} [options={}]
+   *        The key/value store of player options.
+   */
+  constructor(player, options) {
+
+    // pass the inline option down to the VolumeControl if
+    // the VolumeControl is on.
+    if (options.inline === false && (!options.volumeControl || isPlain(options.volumeControl))) {
+      options.volumeControl = options.volumeControl || {};
+      options.volumeControl.inline = options.inline;
+    }
+
+    super(player, options);
+
+    // hide this control if volume support is missing
+    checkVolumeSupport(this, player);
+
+    // when the mouse leaves the VolumePanel area hide the VolumeControl (slider/bar)
+    this.on(['mouseenter', 'touchstart'], () => this.volumeControl.show());
+    this.on(['mouseleave', 'touchend'], () => this.volumeControl.hide());
+
+    // when any child of the VolumePanel gets or loses focus
+    // show/hide the VolumeControl (slider/bar)
+    this.children().forEach((child) => {
+      this.on(child, ['focus'], () => this.volumeControl.show());
+      this.on(child, ['blur'], () => this.volumeControl.hide());
+    });
+  }
+
+  /**
+   * Create the `Component`'s DOM element
+   *
+   * @return {Element}
+   *         The element that was created.
+   */
+  createEl() {
+    let orientationClass = 'vjs-volume-panel-horizontal';
+
+    if (this.options_.inline === false) {
```

Comment:
Yup, just thinking of how to simplify the block.
Maybe just switch to `!this.options.inline`, checking a boolean against `false` is a bit redundant, I think.